### PR TITLE
chore: Allow overriding default and redteam providers globally

### PR DESCRIPTION
**Pull request by sklein12**
1. Setup a global manager to handle the redteam provider. Allows us to override it globally.
2. Allow setting a global completion provider override - https://github.com/promptfoo/promptfoo/pull/2333/files#diff-e9d74f25919e5aca3ef968793552d9198df382ca66225c6efe1954b9cfdcd48d

### CODE REVIEW

**Comment by sklein12** [comment]
@codrabbitai can you review this?

**Comment by sklein12** [comment]
@coderabbitai can you review this?

File: src/redteam/providers/shared.ts [hunk 1]
```diff
@@ -20,8 +20,6 @@ export async function loadRedteamProvider({
   jsonOnly?: boolean;
   preferSmallModel?: boolean;
 } = {}) {
-  // FIXME(ian): This approach only works on CLI, it doesn't work when running via node module.
```
Comment (typpo) [inline src/redteam/providers/shared.ts:23]: did you mean to remove this comment?

**Review by typpo — APPROVED** [review]
looks good

File: src/redteam/providers/shared.ts (hunk 1 above)
Comment (sklein12) [reply src/redteam/providers/shared.ts:23]: yes

**Pull request merged**